### ISSUE TITLE
ABRA versions starting from v2.07 crash with single end data

### ISSUE THREAD

**Issue by mfoll** [mozack/abra2#10]
Hi,

I have been successfully using ABRA 2 on single end data, but versions 2.07 and above crash with this error:
```
java.lang.IllegalStateException: Inappropriate call if not paired read
	at htsjdk.samtools.SAMRecord.requireReadPaired(SAMRecord.java:871)
	at htsjdk.samtools.SAMRecord.getFirstOfPairFlag(SAMRecord.java:929)
	at abra.ReAligner.subsetReads(ReAligner.java:722)
	at abra.ReAligner.processRegion(ReAligner.java:1126)
	at abra.ReAligner.processChromosomeChunk(ReAligner.java:474)
	at abra.ReAlignerRunnable.go(ReAlignerRunnable.java:21)
	at abra.AbraRunnable.run(AbraRunnable.java:20)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
```

Flags in my BAM file are all 0 or 16 as expected for single end data.

If this can help version 2.07 outputs this message just before crashing:
```
INFO	Wed Sep 27 15:51:03 CEST 2017	Waiting for writer threads to complete
INFO	Wed Sep 27 15:51:03 CEST 2017	Finishing: test2.abra.bam
```

**Comment by mozack** [comment]
Thanks for reporting this.  Resolved in v2.10

**Comment by mfoll** [comment]
Just tried to and indeed resolved. Thanks for the super quick fix!
